### DISCUSSION TITLE
vite and chunk size warning

### DISCUSSION THREAD

**Discussion by AmosSzust** [quasarframework/quasar#16956] [General - Components / Directives / etc] [answered]
inside the boot file I imported apexcharts. and then I use it
```
import VueApexCharts from "vue3-apexcharts";

export default boot(({ app }) => {
  app.use(VueApexCharts);
});
```
That is causing this file to be larger than 500kb, 543kb to be exact, so I got the warning

> (!) Some chunks are larger than 500 KiB after minification. Consider:
> - Using dynamic import() to code-split the application
> - Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/guide/en/#outputmanualchunks
> - Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.

I tried adding chunkSizeWarningLimit to quasar.conf.js, build section and I even created a new file called vite.config.js with
```
import { defineConfig } from "vite";

export default defineConfig({
  build: {
    chunkSizeWarningLimit: 600 * 1024,
  },
});
```

but nothing helped, I keep getting the warning about 500kib, what am I missing? Thanks

**Comment by rstoenescu** [comment, marked as answer]
Hi,

Configuring Vite is one thing and the Quasar CLI is another. You need to tell Quasar CLI to extend the generated Vite config:

```js
// quasar.config.js
build: {
  extendViteConf (config) {
    config.build.chunkSizeWarningLimit = 650
  }
```

**Reply by AmosSzust** [reply]
Thank you for the quickest reply/answer

**Reply by yusufkandemir** [reply]
For reference: https://quasar.dev/quasar-cli-vite/handling-vite

**Reply by AmosSzust** [reply]
Thanks, I was looking for quasar and chunkSizeWarningLimit to see if there are references but couldn't find any, if I just look for quasar and vite then the above link is one of the results. Next time I will try wider options when I search.

I also tried bing's copilot and chatgpt but none led me to the right solution.
